code/modules/mob/living/carbon/xenomorph/Abilities.dm

```diff
@@ -354,47 +354,128 @@
 		X.layer = MOB_LAYER
 		to_chat(X, "<span class='notice'>You have stopped hiding.</span>")
 
-/datum/action/xeno_action/emit_pheromones
-	name = "Emit Pheromones (30)"
+/datum/action/xeno_action/toggle_pheromones
+	name = "Open/Collapse Pheromone Options"
 	action_icon_state = "emit_pheromones"
+	plasma_cost = 0
+	var/PheromonesOpen = FALSE //If the  pheromone choices buttons are already displayed or not
+
+/datum/action/xeno_action/toggle_pheromones/can_use_action()
+		return TRUE //No actual gameplay impact; should be able to collapse or open pheromone choices
+
+/datum/action/xeno_action/toggle_pheromones/action_activate()
+	var/mob/living/carbon/Xenomorph/X = owner
+	if(PheromonesOpen)
+		PheromonesOpen = FALSE
+		to_chat(X, "<span class ='xenonotice'>You collapse the pheromone button choices.</span>")
+		for(var/datum/action/path in owner.actions)
+			if(istype(path, /datum/action/xeno_action/pheromones))
+				path.remove_action(X)
+	else
+		PheromonesOpen = TRUE
+		to_chat(X, "<span class ='xenonotice'>You open the pheromone button choices.</span>")
+		var/list/subtypeactions = subtypesof(/datum/action/xeno_action/pheromones)
+		for(var/path in subtypeactions)
+			var/datum/action/xeno_action/pheromones/A = new path()
+			A.give_action(X)
+
+/datum/action/xeno_action/pheromones/emit_recovery //Type casted for easy removal/adding
+	name = "Emit Recovery Pheromones (30)"
+	action_icon_state = "emit_recovery"
 	plasma_cost = 30
 
-/datum/action/xeno_action/emit_pheromones/can_use_action()
+/datum/action/xeno_action/pheromones/emit_recovery/can_use_action()
 	var/mob/living/carbon/Xenomorph/X = owner
 	if(X && !X.is_mob_incapacitated() && !X.lying && !X.buckled && (!X.current_aura || X.plasma_stored >= plasma_cost) && !X.stagger && !X.on_fire) //Can't emit pheromones while on fire!
 		return TRUE
 
-/datum/action/xeno_action/emit_pheromones/action_activate()
+/datum/action/xeno_action/pheromones/emit_recovery/action_activate()
 	var/mob/living/carbon/Xenomorph/X = owner
 	if(!X.check_state())
 		return
+	if(!X.check_plasma(30))
+		return
+	if(X.current_aura == "recovery")
```

Comment:
There's a fair bit of repeated code. You can put it in `/datum/action/xeno_action/pheromones/action_activate()` and call it through a `..()` in the specific redefinitions, such as `/datum/action/xeno_action/pheromones/emit_recovery/action_activate()`

Reply:
Please keep this open for maintainers to keep check on the progress, specially because it's not yet done.

Reply:
Should be good; the pheromones now see if ..() returns TRUE or FALSE and is determined by whenever the plasma check succeeds or not